services/static-webserver/client/source/class/osparc/desktop/preferences/pages/TokensPage.js

```diff
@@ -73,12 +73,17 @@ qx.Class.define("osparc.desktop.preferences.pages.TokensPage", {
       }
 
       const createAPIKeyWindow = new osparc.desktop.preferences.window.CreateAPIKey();
-      createAPIKeyWindow.addListener("finished", keyLabel => {
+      createAPIKeyWindow.addListener("finished", e => {
+        const formData = e.getData();
         const params = {
           data: {
-            "display_name": keyLabel.getData()
+            "display_name": formData["name"]
           }
         };
+        if (formData["expiration"]) {
+          const seconds = parseInt((new Date(formData["expiration"]).getTime() - new Date().getTime()) / 1000);
```

Comment:
Just a small question here. This is running client-side, right? But wouldn't these seconds be used server-side? There might be a time-zone issue? Or not?

Reply:
Ok, crazy question, please ignore. It's seconds, won't matter :-)

Reply:
Everything happens in the frontend's (user's) time zone: the date selection and more importantly, the delta calculation. Furthermore, the web api is already expecting the expiration in seconds.